qiskit/algorithms/linear_solvers/linear_solver.py

```diff
@@ -0,0 +1,112 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""An abstract class for linear systems solvers."""
+
+from abc import ABC, abstractmethod
+from typing import Union, Optional, List, Callable
+import numpy as np
+
+from qiskit import QuantumCircuit
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+from qiskit.result import Result
+
+from qiskit.aqua.algorithms import AlgorithmResult
+from .observables.linear_system_observable import LinearSystemObservable
+
+
+class LinearSolverResult(AlgorithmResult):
+    """A base class for linear systems results.
+
+    The linear systems algorithms return an object of the type ``LinearSystemsResult``
+    with the information about the solution obtained.
+
+    Examples:
+
+    Note:
+    """
+    def __init__(self) -> None:
+        super().__init__()
+
+        # Set the default to None, if the algorithm knows how to calculate it can override it.
+        self._state = None
+        self._observable = None
+        self._euclidean_norm = None
+        self._circuit_results = None
+
+    @property
+    def observable(self) -> Union[float, List[float]]:
+        """return the (list of) calculated observable(s)"""
+        return self._observable
+
+    @observable.setter
+    def observable(self, observable: Optional[Union[float, List[float]]]) -> None:
+        if self._observable is None or observable != self._observable:
+            self._observable = observable
+
+    @property
+    def state(self) -> Union[QuantumCircuit, np.ndarray]:
+        """return either the circuit that prepares the solution or the solution as a vector"""
+        return self._state
+
+    @state.setter
+    def state(self, state: Union[QuantumCircuit, np.ndarray]) -> None:
+        if self._state is None or state != self._state:
```

Comment:
If None is allowed here then the param should be marked Optional and the getter likewise.

Reply:
removed Optional